src/components/Rank.tsx

```diff
@@ -28,18 +31,41 @@ const Rank = ({
   dustState,
   detail,
 }: RankProps) => {
-  const [click, setClick] = useState(true);
-  const show = () => {
-    setClick(!click);
+  const [showDetail, setShowDetail] = useState(true);
+  const navigate = useNavigate();
+  const handleClickShowDetail = () => {
+    setShowDetail(!showDetail);
+  };
+  const handleClickSidoDustDetail = ({
+    dataTime,
+    stationName,
+    pm10Value,
+    pm25Value,
+    pm10Grade,
+    pm25Grade,
+  }: Detail) => {
+    const dustState = (
+      (parseInt(pm10Grade) + parseInt(pm25Grade)) /
+      2
+    ).toString();
+    navigate(`/LocalDetail`, {
+      state: {
+        stationName,
+        fineDust: pm10Value,
+        ultraDust: pm25Value,
+        dustState: dustState,
+        dataTime: dataTime,
+      },
```

Comment:
```suggestion
      state: {
        stationName,
        fineDust: pm10Value,
        ultraDust: pm25Value,
        dustState,
        dataTime,
      },
```

자바스크립트의 Object initializer 개념을 활용하면 더 좋을 것 같아요.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer